mono/utils/mono-threads-coop.c

```diff
@@ -279,7 +279,7 @@ mono_threads_exit_gc_safe_region_unbalanced (gpointer cookie, gpointer *stackdat
 
 	info = (MonoThreadInfo *)cookie;
 
-	check_info (info, "exit", "safe");
+	check_info (info, "exit", "safe", *stackdata);
 
 	switch (mono_threads_transition_done_blocking (info)) {
```

Comment:
We should augment this one, too. so we get a function name on the `BLOCKING->RUNNING` transition.

Reply:
Got it I think. And maybe trace_state_change?